sites/upsun/src/learn/overview/app-life-cycle.md

```diff
@@ -0,0 +1,96 @@
+---
+title: The application lifecycle
+weight: 3
+description: "Understand the {{% vendor/name %}} application lifecycle and learn how to use build, deploy, and runtime hooks to control app behavior"
+---
+
+Hooks let you run custom commands at specific points in your application’s lifecycle during [`build`, `deploy`](/learn/overview/build-deploy.html), or `runtime`. They’re essential for setting up your app, managing graceful shutdowns, or preparing instances to handle traffic in autoscaling environments.
+
+Each web application on {{% vendor/name %}} can define web commands inside its `.upsun/config.yaml` file:
+
+```yaml
+applications:
+  myapp:
+    type: 'python:3.13'
+    web:
+      commands:
+        pre_start: "./scripts/setup.sh"
+        start: "uwsgi --ini conf/server.ini"
+        post_start: "./scripts/warm-up.sh"
+```
+## Web commands
+
+| Name | Type | Required | Description |
+|------|------|-----------|--------------|
+| **pre_start** | string | No | Runs just before `start`. Useful for short setup actions that must run per instance, such as moving cache files or setting permissions. |
+| **start** | string | Yes | The main command that launches your app. If it terminates, {{% vendor/name %}} restarts it immediately. |
+| **post_start** | string | No | Runs after the `start` command but before the container is added to the router. This lets you complete warm-up tasks before the app starts handling traffic. |
+
+{{< note theme="info" title="Note" >}}
+
+For more information about web commands, visit the [Single-runtime Image page](/create-apps/app-reference/single-runtime-image.html#web-commands).
+
+{{< /note >}}
+
+## Lifecycle overview
+
+Every application instance follows the same start-up flow:
+
+- Pre-start prepares each instance before launch.
+- Post-start completes warm-up tasks before the instance is routed into live traffic.
+
+![The steps in the start-up flow](/images/hook-cycle.png "0.50")
+
+This structure ensures applications can start cleanly, scale horizontally, and serve requests reliably, especially in autoscaling environments.
+
+## Use cases 
+
+| Hook | When to use | Example task |
+|------|--------------|----------------|
+| **pre_start** | For configuration and prep that must complete before your app starts | Moving cache, updating file permissions |
+| **post_start** | For initialization after your app starts but before it serves traffic | Cache warming, dependency loading |
+
+### Autoscaling example
+
+When autoscaling [adds new instances](/manage-resources/autoscaling.html#thresholds) in response to traffic spikes, each instance must start and warm up before serving live requests. Use `post-start` to complete initialization tasks such as:
```

Comment:
let's be careful here again... autoscaling does *not* currently react to traffic at all, only to cpu / memory usage